virtual_ecosystem/models/litter/litter_pools.py

```diff
@@ -35,6 +40,17 @@ def calculate_change_in_litter_variables(
     lignin_above_structural: NDArray[np.float32],
     lignin_woody: NDArray[np.float32],
     lignin_below_structural: NDArray[np.float32],
+    deadwood_production: NDArray[np.float32],
```

Comment:
Ok, yes, this is on the long side. 

I wonder if you need this function at all, to start with. In the end, this is a top level function calling a bunch of other functions that do the heavy lifting, and that is all the `update` method above is doing. So you have a top level function (`update`) that just calls another high level function (`calculate_change_in_litter_variables`). I don't see why you cannot put all the contents of this function directly into `update` and save unnecessary, top level functions with long signatures.

Reply:
Another option is to make the litter pools a class, with a reference to data, and rewrite all of these as method that use that data.

```python
class LitterPool:
    def __init__(self, data):
        self.data = data

    def calculate_change_in_litter_variables(self) -> dict:
        ...

    # And other relevant methods
```
And then, in the `LitterModel` just use above:

```python
class LitterModel(...):
    self __init__(...):
        ...
        self.litter_pool = LitterPool(data)

    self.update(...):
        updated_vars = self.litter_pool.calculate_change_in_litter_variables()
        ...
```
Or something along these lines. This is just a sketch.

Reply:
Yeah I do wonder whether the separate function is really warranted. I think I've basically duplicated the structure I've used for the `soil` model, but there the function sets up an integrator, passes the lower level function into the integrator and then extracts the result, whereas here it just calls the lower level function

Reply:
I think I will just delete the lower level function and move everything into `LitterModel.update()` as that seems like the simplest approach. I'll bear the extra class approach in mind when thinking about the `soil` model case where a more complex structure is required